api/src/paths/dwc/eml.ts

```diff
@@ -0,0 +1,921 @@
+      geographicCoverage.datasetGPolygon[i].datasetGPolygonOuterGRing.gRingPoint[g] = {
+        gRingLatitude: point[0],
+        gRingLongitude: point[1]
+      };
+    });
+  });
+
+  return geographicCoverage;
+};
+
+/**
+ * Return funding source eml.
+ *
+ * @param {any[]} fundingSourceRows
+ * @return {Eml}
+ */
+const getFundingEML = (fundingSourceRows: any[]): Eml => {
+  const funding: Eml = { section: [] };
+
+  fundingSourceRows.forEach(function (row: any, i: number) {
+    funding.section[i] = { title: 'Funding Source', para: row.funding_source_name };
+    funding.section[i].section = {
+      title: 'Investment Action Category',
+      para: row.investment_action_category_name,
+      section: [
+        { title: 'Funding Source Project ID', para: row.funding_source_project_id },
+        { title: 'Funding Amount', para: row.funding_amount },
+        { title: 'Funding Start Date', para: new Date(row.funding_start_date).toISOString().split('T')[0] },
+        { title: 'Funding End Date', para: new Date(row.funding_end_date).toISOString().split('T')[0] }
+      ]
+    };
+  });
+
+  return funding;
+};
+
+/**
+ * Return default message if value not provided.
+ *
+ * @param {string} valueToCheck
+ * @param {IDBConnection} connection
+ * @return {string | number}
+ */
+const checkProvided = (valueToCheck: string | number | null): string | number => {
+  // the EML specification requires all fields have values
+  // fail gracefully by providing standard message that data is not supplied
+  const notSuppliedMessage = 'Not Supplied';
+
+  if (valueToCheck === null) {
+    return notSuppliedMessage;
+  }
+
+  return valueToCheck;
+};
+
+/**
+ * Get occurrence submission record associated with data package ID.
+ *
+ * @param {number} data_package_id
+ * @param {IDBConnection} connection
+ * @return {*} {Promise<void>}
+ */
+export const getSurveyOccurrenceSubmission = async (dataPackageId: number, connection: IDBConnection): Promise<any> => {
+  if (!dataPackageId || !connection) {
+    throw new HTTP400('Failed to build SQL update statement');
+  }
+  const sqlStatement = getSurveyOccurrenceSubmissionSQL(dataPackageId);
+
+  if (!sqlStatement) {
+    throw new HTTP400('Failed to build SQL update statement');
+  }
+
+  const response = await connection.query(sqlStatement.text, sqlStatement.values);
+
+  if (!response || response.rowCount != 1) {
+    throw new HTTP400('Failed to acquire distinct survey occurrence submission record');
+  }
+
+  return response.rows[0];
+};
+
+/**
+ * Get data package record associated with data package ID.
+ *
+ * @param {number} data_package_id
+ * @param {IDBConnection} connection
+ * @return {*} {Promise<void>}
+ */
+export const getDataPackage = async (dataPackageId: number, connection: IDBConnection): Promise<any> => {
+  const sqlStatement = getDataPackageSQL(dataPackageId);
+
+  if (!sqlStatement) {
+    throw new HTTP400('Failed to build SQL update statement');
```

Comment:
small nit: these all say `update` statement, maybe update to `get`

Reply:
adjusted